execution-core/src/bytecode.rs

```diff
@@ -0,0 +1,59 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! Wrapper for a long data that we want to keep the integrity of.
+
+extern crate alloc;
+use alloc::vec::Vec;
+use bytecheck::CheckBytes;
+use dusk_bytes::Error as BytesError;
+use dusk_bytes::Error::InvalidData;
+use rkyv::{Archive, Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Archive, Serialize, Deserialize)]
+#[archive_attr(derive(CheckBytes))]
+/// Holds bytes of bytecode and its hash.
+pub struct Bytecode {
+    /// Hash of the bytecode bytes.
+    pub hash: [u8; 32],
+    /// Bytecode bytes.
+    pub bytes: Vec<u8>,
+}
+
+impl Bytecode {
+    /// Provides contribution bytes for an external hash.
+    #[must_use]
+    pub fn to_hash_input_bytes(&self) -> Vec<u8> {
+        self.hash.to_vec()
+    }
+
+    /// Serializes this object into a variable length buffer
+    #[must_use]
+    pub fn to_var_bytes(&self) -> Vec<u8> {
+        let mut bytes = Vec::new();
+        bytes.extend(self.hash);
```

Comment:
Shouldn't be the bytecode vec part of the serialization?

Reply:
I'm saying this, because the byte code need to be part of the transaction that is sent over the network
And iiuc, with current impl only the hash will be part of